### PR TITLE
VZ-9459, VZ-9524.  Initial changes to support use of user-provided Cert-Manager instance

### PR DESCRIPTION
**Pull request by michael-cico**
Changes to allow the use of a user-provided Cert-Manager installation, broadly speaking:
- Move all configuration of Verrazzano Cert-Manager resources to a new component, `cert-manager-config`
- Introduces a new, internal `cert-manager-ocidns-webhook` component to implement OCI DNS challenges without patching the Cert-Manager CRDs
- Explicitly declare the `verrazzano-cluster-issuer` in the ingress annotations for all Verrazzano components that need it (including VMO-managed components)
- Uses an updated VMO instance to set the ClusterIssuer assignment for OS/OSD/Grafana
- Implement cleanup of dangling CM resources on uninstall
- Prevent clobbering of externally installed Cert-Manager on uninstall

The existing `certManager` component functions:
- only manages the installation of the Cert-Manager chart if the component is enabled; all configuration moved to the `cert-manager-config` component
- the webhook validators already check for an existing Cert-Manager installation to avoid clobbering one if it's there by checking the namespace for the Verrazzano annotation

The `cert-manager-ocidns-webhook` implements a webhook contract defined by Cert-Manager; this builds on an implementation by @shihchang last year that was never folded into Verrazzano. Changes
- A new image in the BOM 
- A chart for deploying it into the `cert-manager` namespace
- Deployed only when OCI DNS is configured in the Verrazzano CR
- Behavior preserves the existing OCI DNS integration via an auth secret that the webhook knows how to parse

The `cert-manager-config` component mostly contains code refactored from the current `certManager` component, and acts as an internal component which:
- Is enabled only if it detects the existence of the Cert-Manager CRDs required by Verrazzano
- If enabled, it will create/update the Verrazzano ClusterIssuer and related resources as needed
- Sets up the Verrazzano `ClusterIssuer` as we've previously done, with some changes to accommodate the new OCI DNS webhook configuration

This does not currently address:
- Namespace changes for the Verrazzano Cert-Manager instance
- Namespace changes for Verrazzano-related Cert-Manager resources
- E2E tests, which will be addressed in a follow-up PR

### Testing with an external Cert-Manager

- Apply the CM resources and operators:

```
kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.11.0/cert-manager.yaml
```

- Deploy the VPO

- Apply a configuration with the Verrazzano Cert-Manager instance disabled

```
vz install -f - <<EOF
apiVersion: install.verrazzano.io/v1beta1
kind: Verrazzano
metadata:
  name: my-verrazzano
spec:
  profile: dev
  components:
    certManager:
      enabled: false
EOF
```

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/certmanagerconfig/certmanager_config_component.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+			SupportsOperatorUninstall: true,
+			//ImagePullSecretKeyname:    "global.imagePullSecrets[0].name",
+			MinVerrazzanoVersion: constants.VerrazzanoVersion1_0_0,
+			Dependencies:         []string{networkpolicies.ComponentName, certmanager.ComponentName},
+		},
+	}
+}
+
+// IsEnabled returns true if the cert-manager-config is enabled, which is the default
+func (c certManagerConfigComponent) IsEnabled(_ runtime.Object) bool {
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/certmanagerconfig/certmanager_config_component.go:63]: How will this work if someone wants to disable cert-manager and have us uninstall it?  Do we have a way of knowing that vz installed cert-manager?

Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/certmanagerconfig/certmanager_config_component.go:53]: This is largely a VZ 1.6 issue I think; with a-la-carte we need to distinguish between verrazzano-managed components and others.  @mbenmclean has a solution for this in the E2E testing branch he's working on.

This should not affect the 1.5 backport.

File: platform-operator/thirdparty/charts/cert-manager-webhook-oci/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+appVersion: "1.0"
+description: Allow cert-manager to solve DNS challenges using Oracle Cloud Infrastructure DNS
+name: cert-manager-webhook-oci
+version: "1.0.0"
```
Comment (michael-cico) [inline platform-operator/thirdparty/charts/cert-manager-webhook-oci/Chart.yaml:5]: Chart version update to 1.6.0

File: platform-operator/controllers/verrazzano/component/certmanager/certmanager_component.go [hunk 3]
```diff
@@ -197,36 +197,12 @@ func (c certManagerComponent) PreInstall(compContext spi.ComponentContext) error
 	if err != nil {
 		return log.ErrorfNewErr("Failed to apply the cert-manager manifest: %v", err)
 	}
-	if err := common.ProcessAdditionalCertificates(log, cli, vz); err != nil {
-		return err
-	}
+	//if err := common.ProcessAdditionalCertificates(log, cli, vz); err != nil {
```
Comment (mbenmclean) [inline platform-operator/controllers/verrazzano/component/certmanager/certmanager_component.go:200]: delete commented code

File: platform-operator/controllers/verrazzano/component/certmanagerconfig/certmanager_config_component.go (hunk 1 above)
Comment (mbenmclean) [reply platform-operator/controllers/verrazzano/component/certmanagerconfig/certmanager_config_component.go:53]: Yes I'm addressing this as part of my branch

File: platform-operator/controllers/verrazzano/component/certmanagerocidns/certmanagerocidns_test.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+// GIVEN a call to this fn
+// WHEN I call PreInstall with dry-run = true
+// THEN no errors are returned
+func TestCertManagerOciDNSPreInstallDryRun(t *testing.T) {
+	client := fake.NewFakeClientWithScheme(k8scheme.Scheme)
+	err := fakeComponent.PreInstall(spi.NewFakeContext(client, &vzapi.Verrazzano{}, nil, true))
+	assert.NoError(t, err)
+}
+
+// TestIsCertManagerReadyOciDNS tests the IsReady function with OCI-DNS enabled
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/certmanagerocidns/certmanagerocidns_test.go:71]: Should remove commented code at some point, not a blocker

**Review by pfmackin — DISMISSED** [review]

File: platform-operator/controllers/verrazzano/component/certmanager/certmanager_component.go (hunk 3 above)
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/certmanager/certmanager_component.go]: Fixed

File: platform-operator/thirdparty/charts/cert-manager-webhook-oci/Chart.yaml (hunk 2 above)
Comment (michael-cico) [reply platform-operator/thirdparty/charts/cert-manager-webhook-oci/Chart.yaml]: Fixed

**Review by pfmackin — APPROVED** [review]

File: platform-operator/controllers/verrazzano/component/certmanagerocidns/certmanagerocidns_test.go (hunk 4 above)
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/certmanagerocidns/certmanagerocidns_test.go:111]: done, and added unit tests.

**Pull request merged**